components/board/genericlinux/gpio.go

```diff
@@ -280,6 +280,12 @@ func (pin *gpioPin) Close() error {
 	pin.mu.Lock()
 	defer pin.mu.Unlock()
 
+	// If the entire server is shutting down, it's important to turn off all pins so they don't
+	// continue outputting signals we will no longer control.
+	if err := pin.setInternal(false); err != nil {
+		return err
+	}
```

Comment:
[q] this should always be called before line 289? Or does it not matter when we unexport our definitions before setting states up or down?

Reply:
It doesn't matter, but double-checking that it doesn't matter made me notice that `setInternal` calls `openGpioFd`, which contains a copy of lines 289-294 (when opening the GPIO interface, we close the PWM interface). I'll remove the duplicate code below.